spec/specification/consumer_spec.rb

```diff
@@ -289,6 +289,14 @@ module Pod
         test_consumer.requires_app_host?.should.be.true
       end
 
+      it 'allows to specify a specific app host to use' do
```

Comment:
what happens if we specify an app host name that is not declared within the podspec?

Reply:
if we require `requires_app_host` to be true in order for this flag to function we probably want a test for it.

Reply:
at the moment that checking is done in main cocoapods during integration, wasn't sure the best place to do that validation in core.

Reply:
you can probably perform it at this level check `linter.rb` in Core

Reply:
I believe this is still missing a test case that checks that the value of `app_host_name` actually matches an app spec. If it doesn't we should probably error out.

Reply:
since the app spec can be from another pod, there's no immediate way to validate that

Reply:
wow it can be from another pod?! I did not know this was part of the design. This seems pretty bad imo, how will the validator (lint) deal with this? Do we implicitly add a `s.dependency`?

Reply:
I don't like doing things implicitly, but I am adding a check to verify that we depend on a pod with the right name. If the app host doesn't exist, that'll come up during integration, but I don't know of a way to error out of that earlier.

Reply:
IMO we shouldn't add a dependency implicitly however I don't mind the fully qualified name (i.e. `$PodspecName/$AppSpecName`. One thing that i can see as a gotcha (super edge case) is if a podspec or the app specs has a `/` in the name? I'm not sure if the cocoapods validator doesn't allow that or not however if it does it could make it so there is 2 app specs with the same fully qualified name.

Reply:
> I'm not sure if the cocoapods validator doesn't allow that or not however if it does it could make it so there is 2 app specs with the same fully qualified name.

I do not believe it is allowed for root specs.

Reply:
See https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/specification/linter.rb#L203-L206